src/bugle_forms/model/form.clj

```diff
@@ -18,24 +21,58 @@
   {:form/id (UUID/randomUUID)
    :form/name name
    :form/owner user-id
+   :form/status :draft
    :form/created (Instant/now)})
 
 (defn insert!
   "Insert form in database."
   [form]
-  (sql/insert! db/datasource form-table form))
+  (->> (update form :form/status (comp jdbc-types/as-other name))
+       (sql/insert! db/datasource form-table)))
 
 (defn get
   "Retrieve form details."
   [id]
-  (if-some [form (sql/get-by-id db/datasource form-table id)]
+  (if-some [form (some-> (sql/get-by-id db/datasource form-table id)
+                         (update :form/status keyword))]
     form
-    {:error "Form not found."}))
+    {:error :form-not-found}))
+
+(defn link
+  "Get the share link for a form.
+  Returns `nil` if the form is invalid or unpublished."
+  [form]
+  (when (and (s/valid? ::specs/form form)
+             (= :published (:form/status form)))
+    (str "/form/" (:form/id form))))
 
 (defn get-forms
   "Retrieve forms belonging to a user."
   [{:keys [uuid]}]
-  (plan/select!
-   db/datasource
-   [:id :name :created]
-   ["select id, name, created from form where owner = ? order by created desc" uuid]))
+  (into []
+        (map (fn [row]
+               (-> (select-keys row [:form/id :form/name :form/owner
+                                     :form/status :form/created])
+                   (update :form/status keyword))))
+        (jdbc/plan db/datasource
+                   ["select id, name, owner, status, created
+                    from form where owner = ?
+                    order by created desc" uuid])))
+
+(defn published?
+  "Is the form having `id` published?"
+  [form]
+  (= :published (:form/status form)))
+
+(defn publish!
```

Comment:
Why append a bang here? Wondering what your convention of choice is.

Reply:
Was under the impression that atomic mutation in clojure should end with `!`.  Turns out this is only a thing [for STM stuff](https://guide.clojure.style/#naming-unsafe-functions). I'll change this.

Reply:
Actually I'm not too sure about this now. I have already named all my model function that do database mutations like `insert!` etc.

Maybe this convention was misguided, but I think I shouldn't break the local convention now?